src/server/services/Analytics.js

```diff
@@ -0,0 +1,78 @@
+require('dotenv').config();
+
+// Mongo setup
+const mongourl = process.env.DB_URL;
+const dbName = process.env.DB_DATABASE;
+const client = require('../models/db');
+
+let analytics = {};
+
+/**
+ * Returns Library analytics for a given class.
+ *
+ * @returns {object}   Analytics
+ */
+async function getAnalytics() {
+  await client.connect(mongourl);
+  const dbAnalytics = client.db(dbName);
+  const cursor = await dbAnalytics.collection('analytics').find();
+  const result = await cursor.toArray();
+
+  result.map(x => {
+    x.reserve_clicks =
+      x.reserve_clicks === undefined ? 0 : x.reserve_clicks.length;
+    x.research_clicks =
+      x.research_clicks === undefined ? 0 : x.research_clicks.length;
+    x.lib_tour_clicks =
+      x.lib_tour_clicks === undefined ? 0 : x.lib_tour_clicks.length;
+    x.research_tuts_clicks =
+      x.research_tuts_clicks === undefined ? 0 : x.research_tuts_clicks.length;
+
+    x.total_reserve_clicks =
+      x.total_reserve_clicks === undefined ? 0 : x.total_reserve_clicks;
+    x.total_research_clicks =
+      x.total_research_clicks === undefined ? 0 : x.total_research_clicks;
+    x.total_lib_tour_clicks =
+      x.total_lib_tour_clicks === undefined ? 0 : x.total_lib_tour_clicks;
+    x.total_research_tuts_clicks =
+      x.total_research_tuts_clicks === undefined
+        ? 0
+        : x.total_research_tuts_clicks;
+    return result;
```

Comment:
@finnzink Why does this return the array it is mapping over? Seems strange... sure it shouldn’t be return x?

Reply:
the map function needs to return something (it's an arrow function). If you return x you just return a single element (a single course)